### PR TITLE
Represent event format changes more clearly in the spec

### PR DESCRIPTION
**Pull request by turt2live**
Rendered: see 'docs' status check

### CODE REVIEW

**Comment by turt2live** [comment]
feedback: we should put in a minimal event (types, content, room ID, etc) instead of outright remove it. Descriptions seem fine otherwise.

**Comment by uhoreg** [comment]
image
(also in `GET /_matrix/federation/v1/make_join/{roomId}/{userId}` and `PUT /_matrix/federation/v1/exchange_third_party_invite/{roomId}`)

**Comment by uhoreg** [comment]
(though the one in exchange_third_party_invite is also in the current version, so might not be caused by this patch)

File: api/server-server/joins.yaml [hunk 1]
```diff
@@ -72,96 +76,64 @@ paths:
...
                         type: string
-                        description: Not used.
-                    required:
-                      # Every other field is already flagged as required by the $ref
-                      - state_key
+                        description: The value ``join``.
+                        example: "join"
+                    required: ['membership']
+                required:
+                  - state_key
```
Comment (uhoreg) [inline api/server-server/joins.yaml]: `sender` should also be required

File: api/server-server/leaving.yaml [hunk 2]
```diff
@@ -189,81 +171,58 @@ paths:
...
-                  # Every other field is already flagged as required by the $ref
-                  - state_key
+                    description: The value ``leave``.
+                    example: "leave"
+                required: ['membership']
+              depth:
+                type: integer
+                description: This field must be present but is ignored; it may be 0.
+                example: 12
+            required:
```
Comment (uhoreg) [inline api/server-server/leaving.yaml]: `depth` should also be required

**Pull request merged**